Sources/CBacktrace/fileline.c

```diff
@@ -94,19 +203,28 @@ fileline_initialize (struct backtrace_state *state,
 	  filename = state->filename;
 	  break;
 	case 1:
-	  filename = "/proc/self/exe";
+	  filename = getexecname ();
```

Comment:
@tomerd actually, this will regress the fix in https://github.com/swift-server/swift-backtrace/pull/36 . I think you should probably apply that again.

Reply:
It might be worth pinging the original author @ianlancetaylor. @ianlancetaylor if you have time, I'd be much appreciated if you'd let us know what you think about the changes (see explanation on the PR) in #36.

Reply:
good catch @weissi thanks

Reply:
@ianlancetaylor Thank you for looking into this. I apologise, the description of the PR w.r.t. `getexecname()` is indeed misleading and not wrong, let me run you through my thought process at the time:

- The actual fix for the underlying issue was this: https://github.com/swift-server/swift-backtrace/pull/36/files#diff-3b807e3382126cd3e623b416fa0ac4e79619f79db52cec4cba395daec969ac68R24 . So the crucial fix was to _not_ start looking into `argv[0] (== CommandLine.arguments[0])`.
- Then I looked up what `getexecname()` does (which was unknown to me -- which makes sense because it's not available on Linux) and the manpage says that it essentially returns `argv[0]` so I concluded that this would give us worse information than `/proc/self/exe` and moved it down

Now with your new information that `getexecname()` only exists on Solaris (which in turn won't have `/proc/*`) I do agree that we indeed _do not_ need to apply my changes to libbacktrace as long as we don't lose [the actual fix](https://github.com/swift-server/swift-backtrace/pull/36/files#diff-3b807e3382126cd3e623b416fa0ac4e79619f79db52cec4cba395daec969ac68R24). But that's in the Swift codebase so there's no risk we'll lose it.

Another way of phrasing this would be: On Linux, we'll have `/proc/*` but _not_ `getexecname()`. On Solaris we'll have `getexecname()` but not `/proc/*` so their relative order doesn't matter a bit. I just didn't know that no Linuxes would have `getexecname()` and just did the change because it was easy.

@tomerd tl;dr your PR looks fine as is.